mimic/rest/maas_api.py

```diff
@@ -518,9 +518,21 @@ def delete_entity(self, request, tenant_id, entity_id):
         Delete an entity, all checks that belong to entity, all alarms that belong to those checks
         """
         entities = self._entity_cache_for_tenant(tenant_id).entities_list
+        previous_size = len(entities)
         entities[:] = [entity for entity in entities
                        if entity.id != entity_id]
 
+        if len(entities) == previous_size:
```

Comment:
I just took a look through #412 again. It is true, we have not had a serious bug with iterating and modifying the list, even with the old way the code was written. However, it can lead to confusing code. For instance, you might see:

```python
# Delete an entity
for i in range(len(entities)):
    if entities[i].id == entity_id:
        del entities[i]
        break

for check in checks:
    if check.entity_id == entity_id:
        del checks[checks.index(check)]
```

There are 2 obvious problems with this code that I can think of off the top of my head:

1. It's very easy to look at this code, perhaps from the angle of copy and pasting it into some other delete function, and do this:

    ```python
    for i in range(len(checks)):
        if checks[i].entity_id == entity_id:
            del checks[i] # BUG!
    ```

2. Deleting a check becomes an O(N^2) operation, by having to lookup the index every single time a check that needs to be deleted is found.